File: Practica5/Marco teorico/1. Referencias/B. Auto Referencias/1.2.java
class Clase {
    private int variableEntera;
    private float variableFlotante;

    public Clase() {
        variableEntera = 0;
        variableFlotante = 0;
    }

    public int getVariableEntera() {
        return this.variableEntera;
    }

    public float getVariableFlotante() {
        return this.variableFlotante;
    }

    public Clase sumaEntero(int n) {
        variableEntera += n;
        return this;
    }

    public Clase sumaFlotante(float n) {
        variableFlotante += n;
        return this;
    }
}

public class Main {
    public static void main(String[] args) {
        Clase objeto = new Clase();
        System.out.println(objeto.getVariableEntera());
        System.out.println(objeto.getVariableFlotante());
        objeto.sumaEntero(5).sumaFlotante(3.5f);
        System.out.println(objeto.getVariableEntera());
        System.out.println(objeto.getVariableFlotante());
    }
}

File: Practica5/Marco teorico/1. Referencias/B. Auto Referencias/1.2.py
class Clase:
    def __init__(self):
        self.variableEntera = 0
        self.variableFlotante = 0

    def getVariableEntera(self):
        return self.variableEntera

    def getVariableFlotante(self):
        return self.variableFlotante

    def sumaEntero(self, n):
        self.variableEntera += n
        return self

    def sumaFlotante(self, n):
        self.variableFlotante += n
        return self

objeto = Clase()
print(objeto.getVariableEntera())
print(objeto.getVariableFlotante())
objeto.sumaEntero(5).sumaFlotante(3.5)
print(objeto.getVariableEntera())
print(objeto.getVariableFlotante())
